homeassistant/components/pi_hole/config_flow.py

```diff
@@ -101,25 +104,6 @@ async def async_step_user(
             errors=errors,
         )
 
-    async def async_step_api_key(
```

Comment:
I have removed the API key as a separate step for a few reasons
1. The info in the previous step is not required
2. Due to the ambiguity between V5 and V6 APIs it is not easy to determine connectivity and the API version separately